includes/class-wc-payments-customer-service.php

```diff
@@ -248,37 +238,76 @@ public function clear_cached_payment_methods_for_user( $user_id ) {
 	}
 
 	/**
-	 * Build the customer description string.
+	 * Given a WC_Order or WC_Customer, returns an array representing a Stripe customer object.
+	 * At least one parameter has to not be null.
 	 *
-	 * @param WP_User $user WordPress user.
-	 * @param string  $name Customer's full name.
+	 * @param WC_Order    $wc_order    The Woo order to parse.
+	 * @param WC_Customer $wc_customer The Woo customer to parse.
 	 *
-	 * @return string
+	 * @return array Customer data.
 	 */
-	private function build_description_string( $user, $name ) {
-		if ( $user->ID > 0 ) {
+	public static function map_customer_data( $wc_order = null, $wc_customer = null ) {
+		if ( null === $wc_customer && null === $wc_order ) {
+			return [];
+		}
+
+		// Where available, the order data takes precedence over the customer.
+		$object_to_parse = $wc_order ?? $wc_customer;
+		$name            = $object_to_parse->get_billing_first_name() . ' ' . $object_to_parse->get_billing_last_name();
+		$description     = '';
+		if ( null !== $wc_customer && ! empty( $wc_customer->get_username() ) ) {
 			// We have a logged in user, so add their username to the customer description.
 			// translators: %1$s Name, %2$s Username.
-			return sprintf( __( 'Name: %1$s, Username: %2$s', 'woocommerce-payments' ), $name, $user->user_login );
+			$description = sprintf( __( 'Name: %1$s, Username: %2$s', 'woocommerce-payments' ), $name, $wc_customer->get_username() );
 		} else {
 			// Current user is not logged in.
 			// translators: %1$s Name.
-			return sprintf( __( 'Name: %1$s, Guest', 'woocommerce-payments' ), $name );
+			$description = sprintf( __( 'Name: %1$s, Guest', 'woocommerce-payments' ), $name );
+		}
+
+		$data = [
+			'name'        => $name,
+			'description' => $description,
+			'email'       => $object_to_parse->get_billing_email(),
+			'phone'       => $object_to_parse->get_billing_phone(),
+			'address'     => [
+				'line1'       => $object_to_parse->get_billing_address_1(),
+				'line2'       => $object_to_parse->get_billing_address_2(),
+				'postal_code' => $object_to_parse->get_billing_postcode(),
+				'city'        => $object_to_parse->get_billing_city(),
+				'state'       => $object_to_parse->get_billing_state(),
+				'country'     => $object_to_parse->get_billing_country(),
+			],
+		];
+
+		if ( ! empty( $object_to_parse->get_shipping_postcode() ) ) {
+			$data['shipping'] = [
```

Comment:
That looks inconsistent, even if data are slightly duplicated, will it be possible to introduce `billing` analog (likely instead of `address`)?

Reply:
I see, now. Thanks for clarifying.